File: 2460 apply-operations-to-an-array/solution.go
func applyOperations(nums []int) []int {
        ans:=make([]int,len(nums))
        idx:=0
        for i:=0;i<len(nums)-1;i++ {
            if nums[i]==0 {
                 continue
            }
            if nums[i]==nums[i+1] {
                ans[idx]=nums[i]*2;
                idx++;
                i++;
            } else {
                ans[idx] = nums[i];
                idx++;
            }
        }
        if nums[len(nums)-2] != nums[len(nums)-1]  {
            ans[idx] = nums[len(nums)-1];
        }
        return ans; 
}

File: 2460 apply-operations-to-an-array/solution.cpp
class Solution {
public:
    vector<int> applyOperations(vector<int>& nums) {
        vector<int> ans(nums.size(),0);
        int idx = 0;
        for(int i=0;i<nums.size()-1;i++) {
            if(nums[i]==0) continue;
            if(nums[i]==nums[i+1]) {
                ans[idx++]=nums[i]*2;
                i++;
            } else {
                ans[idx++] = nums[i];
            }
        }
        if(nums[nums.size()-2] != nums[nums.size()-1])   
            ans[idx] = nums[nums.size()-1];
        return ans; 
    }
};
